pkg/volume/cinder/cinder.go

```diff
@@ -467,70 +466,9 @@ func (c *cinderVolumeUnmounter) TearDown() error {
 	return c.TearDownAt(c.GetPath())
 }
 
-// Unmounts the bind mount, and detaches the disk only if the PD
-// resource was the last reference to that disk on the kubelet.
+// Unmounts the bind mount
 func (c *cinderVolumeUnmounter) TearDownAt(dir string) error {
-	if pathExists, pathErr := mount.PathExists(dir); pathErr != nil {
-		return fmt.Errorf("Error checking if path exists: %v", pathErr)
-	} else if !pathExists {
-		klog.Warningf("Warning: Unmount skipped because path does not exist: %v", dir)
-		return nil
-	}
-
-	klog.V(5).Infof("Cinder TearDown of %s", dir)
-	notmnt, err := c.mounter.IsLikelyNotMountPoint(dir)
-	if err != nil {
-		klog.V(4).Infof("IsLikelyNotMountPoint check failed: %v", err)
-		return err
-	}
-	if notmnt {
-		klog.V(4).Infof("Nothing is mounted to %s, ignoring", dir)
-		return os.Remove(dir)
-	}
-
-	// Find Cinder volumeID to lock the right volume
-	// TODO: refactor VolumePlugin.NewUnmounter to get full volume.Spec just like
-	// NewMounter. We could then find volumeID there without probing MountRefs.
-	refs, err := c.mounter.GetMountRefs(dir)
-	if err != nil {
-		klog.V(4).Infof("GetMountRefs failed: %v", err)
-		return err
-	}
-	if len(refs) == 0 {
-		klog.V(4).Infof("Directory %s is not mounted", dir)
-		return fmt.Errorf("directory %s is not mounted", dir)
-	}
-	c.pdName = path.Base(refs[0])
-	klog.V(4).Infof("Found volume %s mounted to %s", c.pdName, dir)
-
-	// lock the volume (and thus wait for any concurrrent SetUpAt to finish)
-	c.plugin.volumeLocks.LockKey(c.pdName)
```

Comment:
@jsafrane I think this locking is used to resolve the race condition between `TearDownAt` and `SetUpAt` functions during the detach operation, this `TearDownAt` function doesn't make detach anymore, is the race condition between TearDownAt and SetUpAt still a valid issue?